packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -556,6 +569,13 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                 return new TreeTreeEntry(key, snapshot);
             }));
 
+        // NOTE: Search blob concept!
```

Comment:
Something to consider:
Take a look at PR #3822, in particular [this chunk](https://github.com/microsoft/FluidFramework/pull/3822/files#diff-0fe7084dd8c13f6b97378ca507e6d949R214)
We need to ensure somehow that any delay-loading scenarios (like one in DataObject) had a chance to finish initialization (and chance to install hooks) before / as we summarize.
Today, there is no way to do so.
One way how PUreDataStoreFactory can do so is to always install "empty" search hook, that does not add any new blobs, but only waits for DataObject to finish initialization and be able to install real hook, I.e. similar to how the code I pointed to does regarding request handling - it awaits first object creation, and only then allows it to do something useful.
If we go that route, we need to ensure that addition to hook list is handled correctly (i.e. after each hook executed, we test if new hooks were added), This is ugly, I hope we can do better here.

Reply:
I think the best way to solve it is to expose some kind 2-state initialization of data FluidDataStoreRuntime. I'll take a look at it.

Reply:
> This is ugly, I hope we can do better here.

Can you clarify which is meant by "this"

Reply:
This part:
```
           const summarize = runtime.summarize.bind(runtime);
           runtime.summarize = async (...args) => {
                await instanceP;
                return summarize(...args);
            };
```
it assumes that there is such function as "summarize" and that it is async. If it's not, it breaks without any compiler warning, I believe.